External likelihood for optimal observables

TommasoArmadillo opened the pull request:
I modified the existing external likelihood modules for the optimal observables. The modifications are done accordingly to the new constructor syntax introduced in [PR#151](https://github.com/LHCfitNikhef/smefit_release/pull/151) of `smefit_release`. 

I also renamed the file `interface_oos.py` -> `interface_oos_fcc.py`.

Finally, I renamed the classes making explicit the collider, e.g. `OptimalWW161` -> `OptimalWWFCC161`.

The new syntax for the runcard is simply: 
```
OptimalWWFCC161:
    path: external_chi2/optimal_observable/interface_oos_fcc.py
```

arossia94 commented:
@TommasoArmadillo , this seems to overlap fully with the latest update of the external likelihood for the OOs in the ILC/CLIC predictions branch. Please, check what has been done first.

TommasoArmadillo commented:
The optimal observables constructor is compatible with [PR#151](https://github.com/LHCfitNikhef/smefit_release/pull/151) of `smefit_release`.

I renamed the FCC file and classes from `FCC` to `FCCee`.

I adjusted the test in `tests/test_optim_obs_fccee.py`.

I rebased on `main` and solved the conflicts. 

Now the PR is ready for review

jacoterh reviewed (approved):
Thanks a lot for this @TommasoArmadillo, looks good to me. I only left a small suggestion.

TommasoArmadillo commented on external_chi2/optimal_observables/interface_oos_fccee.py:
```diff
@@ -10,54 +13,44 @@
         self.datasets = {
-            "{collider}_ww_lepto_240": "invcov_{collider}_ww_leptonic_240.dat",
-            "{collider}_ww_semilep_240": "invcov_{collider}_ww_semilep_240.dat",
+            "{collider}_ww_lepto_161": "invcov_{collider}_ww_leptonic_161.dat",
```
I agree, I modified the files accordingly.
